Docs/partner.md

```diff
@@ -0,0 +1,28 @@
+# Device Manufacturer Guidance
+
+The LoRaWAN starter kit currently is tested on many popular gateways, we run some of those as part of daily CI/CD pipeline to test integrity and performance of our codebase. However, we cannot test each and every gateway out there and hence we have created a process for device manufacturer to support their gateways and get them highlighted in this repo.
+
+## Step by Step Instructions
+
+If you would like to make your gateway highlighted on our GitHub page, we expect that you contribute to our repo in terms of a sample on how to make the gateway work. This approach provides us with validation that things work on the gateway and also allows others using the same Gateway to leverage the code in the sample. To create the sample, follow these steps:
+
+- Go through the [Developer Guidance](/Docs/devguide.md) to clone the repo and make sure everything works in your local dev environment.
+- You may have to create a specific encoder / decoder, we provide a framework for adding your custom encoder. Refer to [Developer Guidance](/Docs/devguide.md) for more details.
+- Make sure everything works with an Azure subscription with Standard Pricing SKU's, for example we do not support the Free Azure IoT Hub SKU.
+- Once you have the code working, we expect that you create a sample for the custom decoder for your gateway that can be used by any other developer to onboard the starter kit for that device. The sample should be created in the [Samples](/Samples) folder and follow the below structure:
+  
+``` 
+.. Samples
+|-------- .. <Device Manufacturer Name>  (Create this folder for company)
+|------------ .. <Device Gateway Model> (Create this folder for every gateway)
+|--------------- .. Sample source code (add you source code here)
+```
+
+Example:
+
+![Decoder](/Docs/Pictures/samples.png)
+
+- We have provided a [sample reference implementation](/Samples/DecoderSample) of a decoder, please refer to this as a template and leverage the [instructions](/Samples/DecoderSample/ReadMe.md) to create implementation of your customer decoder. 
```

Comment:
I think you are mixing two things together. The gateway should be tested to work with our docker packet forwarder (cloned directly from Semtech) and document the reset pin and spidev version. In case that it needs some changes to the packet forwarder they should provide a packet forwarder edge module that containers the necessary changes. 


Decoders are related to leaf devices and not link in anyway to gateway. In this case it is correct and would be nice that end node manufacturers provide decoder samples

Reply:
I think we should distinguish partners who build gateways and want to put our software on it and partners that build leaf LoRa devices and want to provide a payload decoder reference implementation.

Reply:
i updated the doc with section for both, I think both are vita pieces of information and highlight distinct parts of our  kit. check out the updates and let me know if it makes sense

Reply:
Great improvement! I would however highlight the difference even more.
- Gateway manufacturers will test / make our Edge Hub modules work on their gateway.
- LoRa Sensor manufacturers will be providing decoders for their sensors.
The two are very different.

Reply:
ok, make things a little more explicit, let me know if this is clear enough.

Reply:
I am so sorry to be such a pain on this one. Please don't hate me!!
But it still reads to me like Gateway manufacturers will write encoders/decoders.
Also, encoders have nothing to do with our solution. They happen on the LoRa device side and we don't care at all. We shouldn't mention them to prevent confusion.
We need to create a new folder structure / document for Gateway manufacturers to place their insturctions how to set up their hardware to run with our software.
I also talked to @ronniesa briefly about a new repo or new folder in this repo with "no warranty" implementations of decoders for common LoRa sensors that are on the market. I don't think this belongs into the SensorDecoder.sln as this is the naked template for implementing a decoder and should stay like this.